### PR TITLE
ramips: mt7621: add support for Cudy AP1300 Outdoor

### PR DESCRIPTION
**Pull request by DanielRIOT**
Cudy AP1300 Outdoor is an outdoor unit with IEEE802.11ac and IEEE802.11 b/g/n radios

Hardware:
 - SoC: Mediatek MT7621
 - Flash: 16 MiB XM25QH128C
 - RAM: 128 MiB
 - WLAN: 2x2 5GHz (MTY7613BE), 2x2 2.4GHz (MT7603E)
 - Ethernet: 1x 10/100/1000 Mbps LAN
 - Buttons: 1 Reset button, 1 WPS button
 - LEDs: 5x Green (Status, Internet,LAN, Wifi2.4Ghz, Wifi5Ghz)
 - Serial Console: unpopulated header 115200 8n1 (silkscreen on PCB)
 - Power: POE 802.3af (37-57V DC)

Installation:
Installation can only be done after the unlocked firmware has been installed.
Unlocked firmware is available from Cudy on request or may be available on their website
 ( https://www.cudy.com/blogs/faq/openwrt-software-download )
TFTP boot:
Hold RESET for about 5 seconds while powering on and the device will request a "recovery.bin" via TFTP
 - TFTP server at 192.168.1.88, Filename: recovery.bin The unlocked firmware is based on LEDE 17 and 
   does not have LUCI so SSH/SCP sysupgrade should be used to upload the OpenWRT sysupgrade image
 - LEDE IP : 192.168.10.1, username/password: root/admin

Signed-off-by: Daniel de Kock <daniel@riot.network>

### CODE REVIEW

**Comment by musashino205** [comment]
Please add a missing `Signed-off-by: ` line with your full name to the commit message.

File: target/linux/ramips/dts/mt7621_cudy_ap1300.dts [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+			color = <LED_COLOR_ID_GREEN>;
+			label = "wifi2g";
+		};
+
+		wifi5g {
+			gpios = <&gpio 45 GPIO_ACTIVE_LOW>;
+			color = <LED_COLOR_ID_GREEN>;
+			label = "wifi5g";
+		};
+
```
Comment (musashino205) [inline target/linux/ramips/dts/mt7621_cudy_ap1300.dts:72]: Please remove this empty line.

File: target/linux/ramips/image/mt7621.mk [hunk 2]
```diff
@@ -629,6 +629,19 @@ define Device/comfast_cf-ew72-v2
...
+define Device/cudy_ap1300
+  $(Device/dsa-migration)
+  $(Device/uimage-lzma-loader)
+  DEVICE_VENDOR := Cudy
+  DEVICE_MODEL := AP1300
+  IMAGE_SIZE := 15872k
+  UIMAGE_NAME := R39
+  DEVICE_PACKAGES := kmod-mt7603 kmod-mt7615e kmod-mt7663-firmware-ap \
+	-uboot-envtools
+  SUPPORTED_DEVICES += cudy,ap1300 R39
```
Comment (musashino205) [inline target/linux/ramips/image/mt7621.mk:641]: `cudy,ap1300` is unnecessary here.

File: target/linux/ramips/dts/mt7621_cudy_ap1300.dts [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+		led_sys:status {
+			gpios = <&gpio 4 GPIO_ACTIVE_LOW>;
+			color = <LED_COLOR_ID_GREEN>;
+			label = "status";
```
[the remark this answers is not in the document]
Comment (musashino205) [reply target/linux/ramips/dts/mt7621_cudy_ap1300.dts:46]: And, `LED_FUNCTION_STATUS` is available for `function` property in this LED.

File: target/linux/ramips/dts/mt7621_cudy_ap1300.dts [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+		};
+
+		internet {
+			gpios = <&gpio 3 GPIO_ACTIVE_LOW>;
+			color = <LED_COLOR_ID_GREEN>;
+			label = "internet";
```
Comment (musashino205) [inline target/linux/ramips/dts/mt7621_cudy_ap1300.dts:52]: `function = LED_FUNCTION_WAN_ONLINE;`

File: target/linux/ramips/dts/mt7621_cudy_ap1300.dts [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+		};
+
+		lan {
+			gpios = <&gpio 46 GPIO_ACTIVE_LOW>;
+			color = <LED_COLOR_ID_GREEN>;
+			label = "lan";
```
Comment (musashino205) [inline target/linux/ramips/dts/mt7621_cudy_ap1300.dts:58]: `function = LED_FUNCTION_LAN;`

File: target/linux/ramips/dts/mt7621_cudy_ap1300.dts [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+		};
+
+		wifi2g {
+			gpios = <&gpio 47 GPIO_ACTIVE_LOW>;
+			color = <LED_COLOR_ID_GREEN>;
+			label = "wifi2g";
```
Comment (musashino205) [inline target/linux/ramips/dts/mt7621_cudy_ap1300.dts:64]: `function = LED_FUNCTION_WLAN_2GHZ;`

File: target/linux/ramips/dts/mt7621_cudy_ap1300.dts [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+		};
+
+		wifi5g {
+			gpios = <&gpio 45 GPIO_ACTIVE_LOW>;
+			color = <LED_COLOR_ID_GREEN>;
+			label = "wifi5g";
```
Comment (musashino205) [inline target/linux/ramips/dts/mt7621_cudy_ap1300.dts:70]: `function = LED_FUNCTION_WLAN_5GHZ;`

File: target/linux/ramips/dts/mt7621_cudy_ap1300.dts (hunk 3 above)
[the remark this answers is not in the document]
Comment (DanielRIOT) [reply target/linux/ramips/dts/mt7621_cudy_ap1300.dts]: Thank you for the comments, I have changed the files

File: target/linux/ramips/dts/mt7621_cudy_ap1300.dts (hunk 1 above)
Comment (DanielRIOT) [reply target/linux/ramips/dts/mt7621_cudy_ap1300.dts]: empty line has been removed

**Comment by neheb** [comment]
mt7612 . I think you meant mt7621

File: target/linux/ramips/dts/mt7621_cudy_ap1300.dts [hunk 8]
```diff
@@ -0,0 +1,221 @@
...
+};
+
+&spi0 {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <10000000>;
+		m25p,chunked-io = <0x20>;
```
Comment (hauke) [inline target/linux/ramips/dts/mt7621_cudy_ap1300.dts:91]: This option is not supported by OpenWrt any more, see a66449c60893a168fa0c13a29127d29314b2e348 and bcf4a5f474d1899af65f0974914f737bf9a8d9dd. I think you can just remove it.

**Comment by hauke** [comment]
Please squash the fixup commits into the original one.

File: target/linux/ramips/dts/mt7621_cudy_ap1300.dts (hunk 8 above)
Comment (DanielRIOT) [reply target/linux/ramips/dts/mt7621_cudy_ap1300.dts]: I have removed the chunked IO setting ( it was in the DTS i pulled from the manufacturer's LEDE 17 image )

**Comment by hauke** [comment]
Similar PR was merged in 8fb644e9e8416df0d7051d8e35b57b06a030c05e using #20268

**Pull request closed without merging**